packages/grafana-runtime/src/utils/DataSourceWithBackend.ts

```diff
@@ -44,29 +58,35 @@ export class DataSourceWithBackend<
         throw new Error('Unknown Datasource: ' + q.datasource);
       }
       return {
-        ...this.applyTemplateVariables(q),
         datasourceId: ds.id,
+        datasourceName: q.datasource,
+        refId: q.refId,
+        from: rangeFrom,
+        to: rangeTo,
         intervalMs,
         maxDataPoints,
-        orgId,
+        model: {
+          ...this.applyTemplateVariables(q),
+          datasourceId: ds.id,
+        },
```

Comment:
Breaking change for test datasource since it deserialize model and use intervalMs, maxDataPoints, to, from instead of using each field provided in proto request.